### ISSUE TITLE
Unexpected diff result returned: '4'

### ISSUE THREAD

**Issue by ariffjeff** [desktop/desktop#10640]
### Describe the bug
Attempting to commit some line changes in a file where only some number of lines (out of the total amount of lines changed) are highlighted to be committed causes this error: `Unexpected diff result returned: '4'`.

I can commit _all_ of the changes in the file at the same time by leaving every line highlighted by default, but leaving a single line un-highlighted leads to this error.

I can only reproduce this for a single file... and I've made enough changes in this file that I get the typical message of `The diff is too large to be displayed by default...` so perhaps this has something to do with it.

### Version & OS
Windows 10
Desktop: Version 2.5.5

### Steps to reproduce the behavior
Make a large amount of changes, 2 hundred lines or so in a file.
Try to commit only a line or so.
The error should appear.

### Expected behavior
Commit only highlighted lines.

### Logs
The only addition to the log that appears when I reproduce this error is this line: `2020-09-24T20:52:18.331Z - info: [ui] [Timing] Action 'create commit' for 'onelooksigns/illustrator_extendscripts' took 0.120s`

**Comment by tierninho** [comment]
@ariffjeff Thanks for the report. I tried to reproduce this but I could not.

Any chance this was a binary file? Was the commit part of a merge conflict?

**Comment by ariffjeff** [comment]
@tierninho 
This is not a binary file, it is a plain old javascript file, well in this case an Extendscript file which is just ES3 javascript and the file extension of `.jsx`. 

I don't believe it was part of a merge conflict. With my limited understanding of git all I was doing was committing line changes for this file in the master branch (the only branch in the repo) if that is what you mean. I am the only person working on this repo, although I have used two accounts by accident to commit to this repo in the past but this was before I even created this file recently. So I'm not entirely sure what would be causing this issue.

I was having some possibly related (i doubt it though) git shenanigans beforehand as well **(on a completely different file):** trying to rename files by only changing the first letter of each filename from uppercase to lowercase, committing this change, and then getting problems where git thought I had two files for a single file when I made changes in it. To be more specific I did the following...

1. rename `Abc.jsx` to `abc.jsx`
2. set ignore case sensitivity checking to false
3. commited the file rename. (this appeared in Github Desktop as the creation of `abc.jsx`, AND NOT the deletion of `Abc.jsx`)
4. made a line change in abc.jsx
5. GitHub Desktop would show identical changes in two files: `Abc.jsx` and `abc.jsx` even though only `abc.jsx` exists in the directory.

Again, probably not related. But it's suspicious git behavior non the less at the same time I'm getting this other error.
